bbq/vm/native_functions.go

```diff
@@ -194,4 +191,57 @@ func init() {
 			),
 		)
 	}
+
+	// Register type-bound functions that are common to many types.
+	for _, builtinType := range sema.AllBuiltinTypes {
+		registerBuiltinTypeBoundFunctions(string(builtinType.ID()))
+	}
+
+	derivedTypeQualifiers := []string{
+		commons.TypeQualifierArray,
+		commons.TypeQualifierDictionary,
+
+		// TODO: add other types. e.g; Optional, etc
+	}
+
+	for _, builtinType := range derivedTypeQualifiers {
+		registerBuiltinTypeBoundFunctions(builtinType)
+	}
+}
+
+func registerBuiltinTypeBoundFunctions(typeQualifier string) {
+	for _, boundFunction := range builtinTypeBoundFunctions() {
+		RegisterTypeBoundFunction(
+			typeQualifier,
+			boundFunction,
+		)
+	}
+}
+
+func builtinTypeBoundFunctions() []NativeFunctionValue {
+	return []NativeFunctionValue{
```

Comment:
Is there a particular reason this is a function that returns a new slice, instead of just being a global variable?